src/Scripting/Core/ScriptOptions.cs

```diff
@@ -282,5 +301,17 @@ private static MetadataReference CreateReferenceFromAssembly(Assembly assembly)
         /// <exception cref="ArgumentNullException"><paramref name="imports"/> is null or contains a null reference.</exception>
         public ScriptOptions AddImports(params string[] imports) =>
             AddImports((IEnumerable<string>)imports);
+
+        /// <summary>
+        /// Creates a new <see cref="ScriptOptions"/> with debugging information enabled.
+        /// </summary>
+        public ScriptOptions WithEmitDebugInformation(bool emitDebugInformation) =>
+            emitDebugInformation == EmitDebugInformation ? this : new ScriptOptions(this) { EmitDebugInformation = emitDebugInformation };
+
+        /// <summary>
+        /// Creates a new <see cref="ScriptOptions"/> with specified <see cref="FileEncoding"/>.
+        /// </summary>
+        public ScriptOptions WithFileEncoding(Encoding encoding) => 
```

Comment:
> There are multiple suppressions of this error already

That's not good.  When did these pop up?  Never went over any of these during API reviews.  If this is happening we need to start flagging them and reviewing.